testdata/generic_providers_yaml/fogg.yml

```diff
@@ -48,7 +48,8 @@ envs:
         custom_provider: false
         config:
           baz_token: prod_token_arn
-          aws_assume_role: "TerraformExecutionRole"
+          assume_role: "TerraformExecutionRole"
+          session_name: "foo"
     components:
```

Comment:
as suggested by @zackhee997  - should we instead expect

```yaml
assume_role:
  role: "TerraformExecutionRole"
  session_name: ...
```

and resolver special case to combine `role` value with `AwsConfig` accountID / ...